### PR TITLE
feat: vertically align inline inputs

### PR DESCRIPTION
**Pull request by dmytrokirpa**
Aligns vertically inline inputs such as `Combobox`, `Dropdown`, `Input`, `SpinButton`, and `Textarea` with other components like `Button`.

Addresses issue #30307.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">76</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/Avatar-mount_base.html">633</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/Avatar-mount.html">625</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/Button-mount_base.html">302</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/Button-mount.html">298</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/Field-mount_base.html">1085</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/Field-mount.html">1118</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProvider-mount_base.html">700</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProvider-mount.html">716</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-mount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-mount.html">88</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">36</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">32</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">76</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/MakeStyles-mount_base.html">875</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/MakeStyles-mount.html">876</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/Persona-mount_base.html">1729</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/Persona-mount.html">1747</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/SpinButton-mount_base.html">1403</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/SpinButton-mount.html">1352</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/SwatchPicker-mount_base.html">1635</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31812/perf-test-react-components/SwatchPicker-mount.html">1664</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.094 MB`<br />`270.62 kB` | `1.094 MB`<br />`270.69 kB` | `184 B` <br />`70 B` |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr> | `223.007 kB`<br />`63.137 kB` | `223.035 kB`<br />`63.141 kB` | `28 B` <br />`4 B` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr> | `27.987 kB`<br />`9.446 kB` | `28.015 kB`<br />`9.456 kB` | `28 B` <br />`10 B` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr> | `27.723 kB`<br />`10.131 kB` | `27.775 kB`<br />`10.162 kB` | `52 B` <br />`31 B` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr> | `36.049 kB`<br />`11.835 kB` | `36.102 kB`<br />`11.85 kB` | `53 B` <br />`15 B` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr> | `26.521 kB`<br />`9.741 kB` | `26.572 kB`<br />`9.769 kB` | `51 B` <br />`28 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.741 kB`<br />`60.957 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `104.384 kB`<br />`34.782 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/2384048419ab2fe4fe5ca924f2e9875a195cb723'>2384048419ab2fe4fe5ca924f2e9875a195cb723</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-input/library/src/components/Input/useInputStyles.styles.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ const useRootClassName = makeResetStyles({
   borderRadius: tokens.borderRadiusMedium, // used for all but underline
```
Comment (fabricteam) [inline packages/react-components/react-input/library/src/components/Input/useInputStyles.styles.ts:45]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31812&commitId=ce03a369a3b21a0474b2d0a0568cd9d6fdc9324e&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|0| Removed|
|Avatar Converged.basic.normal.chromium.png|0| Removed|

</details>


  <details><summary><b>Card Converged - Selectable</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged - Selectable.appearance selectable - Outline - Dark Mode.click.chromium.png|0| Added|
|Card Converged - Selectable.appearance selectable - Outline - Dark Mode.hover.chromium.png|0| Added|
|Card Converged - Selectable.appearance selectable - Outline - Dark Mode.normal.chromium.png|0| Added|
|Card Converged - Selectable.appearance selectable - Outline - Dark Mode.selected.chromium.png|0| Added|

</details>


  <details><summary><b>Input Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Input Converged.Inline.default.chromium.png|271| Changed|

</details>


  <details><summary><b>SearchBox Converged</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SearchBox Converged.Size- small.hover.chromium.png|233| Changed|
|SearchBox Converged.Size- small.focused.chromium.png|233| Changed|
|SearchBox Converged.Size- small.default.chromium.png|217| Changed|

</details>


  <details><summary><b>SwatchPicker Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.default.default.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-combobox/library/src/components/Combobox/useComboboxStyles.styles.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ const useStyles = makeStyles({
     justifyContent: 'space-between',
     minWidth: '250px',
     position: 'relative',
+    verticalAlign: 'middle',
```
Comment (spmonahan) [inline packages/react-components/react-combobox/library/src/components/Combobox/useComboboxStyles.styles.ts:36]: Any ideas as to why this one line change increases the bundle size for this component so much?

Comment (dmytrokirpa) [reply packages/react-components/react-combobox/library/src/components/Combobox/useComboboxStyles.styles.ts:36]: I don't think it's related to the current PR, the bundle size metrics appear to be fine after rebasing with the latest master.

Screenshot 2024-07-02 at 12 52 10

**Review by spmonahan — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Pull request merged**